src/sonic-config-engine/minigraph.py

```diff
@@ -915,7 +918,9 @@ def parse_xml(filename, platform=None, port_config_file=None, asic_name=None, hw
     if sub_role is not None:
         current_device['sub_role'] = sub_role
         results['DEVICE_METADATA']['localhost']['sub_role'] =  sub_role
+    if asic_name is not None:
```

Comment:
asic_name was previously added under sub_role. Was it intentional? @SuvarnaMeenakshi , @abdosi ?

Reply:
asic_name is passed to create configuration for asic and only for asic configuration, sub_role is defined.
So if sub_role isNone, asic_name is also none, hence specific check was not added.